src/odemis/driver/focustracker.py

```diff
@@ -21,206 +21,368 @@
 """
 from __future__ import division
 
+from can import CanError
+import can
+from canopen import node
+import canopen  # TODO add to project requirements.
+from canopen.nmt import NmtError
+import logging
 import numbers
+from odemis import model, util
+import os
 import pkg_resources
+import random
+import time
 
-import can
-import canopen  # TODO package canopen and add to project requirements.
 
-from odemis import model
-from odemis.model import HwError
+# The main objects that could be of interest are:
+# 0x6100 AI Input FV  # Field Value: location of the laser spot on the linear CCD (in px)
+# 0x6126 AI Scaling Factor  # Conversion ratio from Input FV to Input PV (µm/px)
+# 0x6130 AI Input PV  # Process Value: location converted to focus position (unit: µm)
+# 0x6138 AI Tare zero  # Net PV = Input PV - Tare zero (unit: µm)
+# 0x6139 AI Autotare  # write 0x61726174 ('tara' in hex), to set Tare zero to the current value
+# 0x6140 AI Net PV  # Net PV = Input PV - Tare zero (unit: µm)
 
-TARGET_POSITION_RANGE = (0, 100e-6)
+POS_SDO = "AI Net PV"
 
+# Expected values, to compare to what is returned by the device
+VENDOR_ID = 0xc0ffee  # Not an official ID, but tasty enough for our internal use
+PRODUCT_CODE = 0x0001
 
-class FocusTrackerCO(model.HwComponent):
+# The maximum acceptable duration since a position update.
+# If the latest known position is "older", the getter will explicitly position.
+MAX_POS_AGE = 0.1  # s
 
-    def __init__(self, name, role, channel, node_idx, **kwargs):
-        """
 
-        channel (str): channel name of can bus
+class FocusTrackerCO(model.HwComponent):
+    """
+    Driver for the in-house (Delmic) focus tracker device.
+    It's connected via CANopen.
+    """
+    def __init__(self, name, role, channel, node_idx, datasheet=None, inverted=None, ** kwargs):
+        """
+        channel (str): channel name of can bus (eg, "can0"). Use "fake" for a simulator.
         node_idx (int): node index of focus tracker
+        datasheet (str or None): absolute or relative path to .dcf configuration file
+          This can be used to set default parameters value. If None, it will use the
+          default .eds file.
+        inverted (set of str): names of the axes which are inverted (IOW, either
```

Comment:
We will probably never have more than just the z-axes, right? It would be nice to extend this comment to mention that for now we only have the z-axes.